File: auto_weibo.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
	"path/filepath"
	"time"
)

type HotTopic struct {
	Word string `json:"word"`
}

type HotTopicsResponse struct {
	Data struct {
		Realtime []HotTopic `json:"realtime"`
	} `json:"data"`
}

func getHotTopics() ([]HotTopic, error) {
	resp, err := http.Get("https://weibo.com/ajax/side/hotSearch")
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}

	var hotTopicsResponse HotTopicsResponse
	err = json.Unmarshal(body, &hotTopicsResponse)
	if err != nil {
		return nil, err
	}

	return hotTopicsResponse.Data.Realtime, nil
}

func generateMarkdownContent(hotTopics []HotTopic) string {
	markdownContent := "# 微博热点话题-go语言生成\n\n"
	for _, topic := range hotTopics {
		word := topic.Word
		link := fmt.Sprintf("https://s.weibo.com/weibo?q=%s", word)
		markdownContent += fmt.Sprintf("- [%s](%s)\n", word, link)
	}

	return markdownContent
}

func addHotTopicsToRepository(hotTopics []HotTopic) error {
	currentTime := time.Now()
	dateStr := currentTime.Format("200601")
	folderName := filepath.Join("weibo", dateStr)
	err := os.MkdirAll(folderName, os.ModePerm)
	if err != nil {
		return err
	}

	fileName := fmt.Sprintf("hot_topics_%s.md", currentTime.Format("150405"))
	filePath := filepath.Join(folderName, fileName)
	markdownContent := generateMarkdownContent(hotTopics)

	err = ioutil.WriteFile(filePath, []byte(markdownContent), os.ModePerm)
	if err != nil {
		return err
	}

	return nil
}

func main() {
	hotTopics, err := getHotTopics()
	if err != nil {
		fmt.Println("Failed to get hot topics:", err)
		return
	}

	err = addHotTopicsToRepository(hotTopics)
	if err != nil {
		fmt.Println("Failed to add hot topics to repository:", err)
		return
	}

	fmt.Println("Hot topics added to repository successfully!")
}

File: auto_weibo.java
package com.weibo;
import java.io.FileWriter;
import java.io.IOException;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.List;

import com.google.gson.Gson;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class Main {

    public static class HotTopic {
        private String word;

        public String getWord() {
            return word;
        }
    }

    public static class HotTopicsResponse {
        private List<HotTopic> realtime;

        public List<HotTopic> getRealtime() {
            return realtime;
        }
    }

    public static void main(String[] args) {
        OkHttpClient client = new OkHttpClient();
        Request request = new Request.Builder()
                .url("https://weibo.com/ajax/side/hotSearch")
                .build();

        try (Response response = client.newCall(request).execute()) {
            if (response.isSuccessful()) {
                Gson gson = new Gson();
                HotTopicsResponse hotTopicsResponse = gson.fromJson(response.body().string(), HotTopicsResponse.class);
                List<HotTopic> hotTopics = hotTopicsResponse.getRealtime();

                if (hotTopics != null && !hotTopics.isEmpty()) {
                    addHotTopicsToRepository(hotTopics);
                }
            } else {
                System.out.println("Failed to get hot topics: " + response.code());
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private static String generateMarkdownContent(List<HotTopic> hotTopics) {
        StringBuilder markdownContent = new StringBuilder("# 微博热点话题\n\n");
        for (HotTopic topic : hotTopics) {
            String word = topic.getWord();
            String link = "https://s.weibo.com/weibo?q=" + word;
            markdownContent.append("- [").append(word).append("](").append(link).append(")\n");
        }

        return markdownContent.toString();
    }

    private static void addHotTopicsToRepository(List<HotTopic> hotTopics) {
        LocalDateTime currentTime = LocalDateTime.now();
        String dateStr = currentTime.format(DateTimeFormatter.ofPattern("yyyyMM"));
        String folderName = "weibo" + File.separator + dateStr;
        new File(folderName).mkdirs();

        String fileName = "hot_topics_" + currentTime.format(DateTimeFormatter.ofPattern("HHmmss")) + ".md";
        String filePath = folderName + File.separator + fileName;
        String markdownContent = generateMarkdownContent(hotTopics);

        try (FileWriter writer = new FileWriter(filePath)) {
            writer.write(markdownContent);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
